### ISSUE TITLE
[Slack] Pool members not being deleted properly

### ISSUE THREAD

**Issue by ansmithf5** [F5Networks/f5-appsvcs-extension#71]
##### Do you already have an issue opened with F5 support?
Github Issues are consistently monitored by F5 staff, but should be considered as best effort only and you should not expect to receive the same level of response as provided by F5 Support. Please [open an case](https://support.f5.com/csp/article/K2633) with F5 if this is a critical issue.

##### ISSUE TYPE

 - Bug Report

##### AS3 BUILD/ VERSION

```
3.9.0 build 1
```

##### BIGIP VERSION

```
13.1.0
```

##### SUMMARY

An issue came in on Slack by a Russell Milliner. Feel free to contact him when this bug is fixed.

When trying to PATCH a pool member with an adminState of Offline the following error is thrown:
```
"results": [
        {
            "message": "declaration failed",
            "response": "A port number or service name is missing for session. Please specify a port number or service name using the syntax session:<port>.",
            "code": 422,
            "host": "localhost",
            "tenant": "tenant1",
            "runTime": 1108
        }
    ]
```

This is because in 'ltm pool' section of map_cli if a pool member has a property deleted, the entire member will be deleted first and then rebuilt. This systems works well, but how it determines the member is flawed. Currently, it just pops the end of the path off and uses that, however that is not always the member.

What needs to happen is to pull the index of the .path that has "members" in it. Then use that index++ to determine the name of the member.

##### STEPS TO REPRODUCE

Basic setup:
```
{
   "class": "AS3",
   "targetPort":8443,
   "declaration":{
      "class":"ADC",
      "schemaVersion":"3.6.0",
      "id":"TEST_Service_HTTP",
      "target":{
         "hostname":"as3.test.com"
      },
      "controls":{
         "class":"Controls",
         "trace":true,
         "logLevel":"debug"
      },
      "Generic_Ten":{
         "class":"Tenant",
         "Generic_App":{
            "class": "Application",
            "template": "generic",
            "generic_Pool": {
               "class": "Pool",
               "members":[
                  {
                     "enable":true,
                     "servicePort":80,
                     "serverAddresses":[
                        "10.128.0.9"
                     ]
                  }
               ]
            }
         }
      }
   }
}
```

To cause failure:
```
[
    {
        "op": "add",
        "path": "/tenant1/app1/pool1/members/0/adminState",
        "value": "offline"
    }
]
```

##### EXPECTED RESULTS

This should run successfully and return a success.

```
"results": [
    {
        "message": "success",
        "lineCount": 17,
        "code": 200,
        "host": "localhost",
        "tenant": "tenant1",
        "runTime": 1128
    }
]
```

##### ACTUAL RESULTS

```
"results": [
        {
            "message": "declaration failed",
            "response": "A port number or service name is missing for session. Please specify a port number or service name using the syntax session:<port>.",
            "code": 422,
            "host": "localhost",
            "tenant": "tenant1",
            "runTime": 1108
        }
    ]
```

**Comment by kaustriaf5** [comment]
Tracked internally as 928

**Comment by dstokesf5** [comment]
This was resolved in 3.10.0. Closing this issue.
